src/mirth/elab.mth

```diff
@@ -625,10 +625,16 @@ def(elab-op-fresh-sig!, Op -- Subst OpSig,
         OP_COERCE -> elab-coerce-sig!,
         OP_LABEL_PUSH -> elab-label-push-sig!,
         OP_LABEL_POP -> elab-label-pop-sig!,
+	OP_DATA_GET_TAG -> elab-data-get-tag-sig!,
         OP_DATA_GET_LABEL -> data-get-label-type freshen-sig OPSIG_APPLY,
         OP_DATA_SET_LABEL -> data-set-label-type freshen-sig OPSIG_APPLY,
     ))
 
+def(elab-data-get-tag-sig!, Data -- OpSig, 
+    MetaVar.new! STMeta swap TData STCons
+    MetaVar.new! STMeta TYPE_INT STCons
```

Comment:
this isn't quite right:

- the stack metavars should be the same
- the data type may take type arguments.

this is just like computing the "tag" type signature, except with fresh metavars. Probably the cleanest approach is to have a function that computes a simple type signature, use it in both places, and call `freshen-sig` on the result, like we do with `data-get-label-type` and `data-set-label-type` above.